File: assignments/assignment-2-solution/src-csharp/TeaShop/Domain/Query/AllInventoryQuery.cs
using Assignment2Solution.Domain.Inventory;

namespace Assignment2Solution.Domain.Query;

/// <summary>
///     The base query that retrieves all items from the inventory repository.
/// </summary>
/// <param name="repository">The inventory repository to query.</param>
public sealed class AllInventoryQuery(InventoryRepository repository) : IInventoryQuery
{
    private readonly InventoryRepository
        _repository = repository ?? throw new ArgumentNullException(nameof(repository));

    /// <inheritdoc />
    public IReadOnlyList<InventoryItem> Execute()
    {
        return _repository.Get();
    }

    /// <inheritdoc />
    public IReadOnlyList<string> AppliedFiltersAndSorts { get; } = Array.Empty<string>();
}

File: assignments/assignment-2-solution/src-java/src/main/java/assignment2solution/domain/inventoryquery/AllInventoryQuery.java
package assignment2solution.domain.inventoryquery;

import assignment2solution.domain.inventory.InventoryItem;
import assignment2solution.domain.inventory.InventoryRepository;

import java.util.List;
import java.util.Objects;

/**
 * The base query that retrieves all items from the inventory repository.
 */
public final class AllInventoryQuery implements IInventoryQuery {
    private final InventoryRepository repository;

    /**
     * Initializes a new instance of the {@link AllInventoryQuery} class.
     */
    public AllInventoryQuery(InventoryRepository repository) {
        this.repository = Objects.requireNonNull(repository, "repository");
    }

    @Override
    public List<InventoryItem> execute() {
        return repository.get();
    }

    @Override
    public List<String> getAppliedFiltersAndSorts() {
        return List.of();
    }
}
